File: Proyectos Android/RetrofitTuto/app/src/main/java/com/example/retrofittuto/entities/PokemonViewModel.java
package com.example.retrofittuto.entities;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.google.gson.FieldNamingPolicy;
import com.google.gson.GsonBuilder;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class PokemonViewModel extends AndroidViewModel {
    public MutableLiveData<ApiResponse> responseMutableLiveData = new MutableLiveData<>();
    private static final Retrofit retrofit = new Retrofit.Builder()
            .baseUrl("https://pokeapi.co/api/v2/")
            .addConverterFactory(GsonConverterFactory.create())
            .build();
    public static ApiClient api = retrofit.create(ApiClient.class);

    public PokemonViewModel(@NonNull Application application) {
        super(application);
    }

    public void search(String offset){
        Call<ApiResponse> response = api.getPokemons(20, Integer.parseInt(offset));
        response.enqueue(new Callback<>() {
            @Override
            public void onResponse(Call<ApiResponse> call, Response<ApiResponse> response) {
                if(response.isSuccessful()){
                    responseMutableLiveData.postValue(response.body());
                }
            }

            @Override
            public void onFailure(Call<ApiResponse> call, Throwable t) {
                Log.d("Error", "Error");
            }
        });
    }
}

File: Proyectos Android/RetrofitTutoKotlin/app/src/main/java/com/example/retrofittutokotlin/entities/PokemonViewModel.kt
package com.example.retrofittutokotlin.entities

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class PokemonViewModel(application: Application) : AndroidViewModel(application) {

    val responseMutableLiveData  = MutableLiveData<ApiResponse>()

    val retrofit = Retrofit.Builder()
        .baseUrl("https://pokeapi.co/api/v2/")
        .addConverterFactory(GsonConverterFactory.create())
        .build()
    val api = retrofit.create(ApiClient::class.java)

    fun search(offset : String){
        val response = api.getPokemons(20, offset.toInt())

        response.enqueue(object : Callback<ApiResponse>{
            override fun onResponse(call: Call<ApiResponse>, response: Response<ApiResponse>) {
                if(response.isSuccessful){
                    responseMutableLiveData.postValue(response.body())
                }
            }

            override fun onFailure(call: Call<ApiResponse>, t: Throwable) {
                Log.e("POKEMON", "Api response error")
            }

        })
    }

}